crates/spawner/src/destroyer.rs

```diff
@@ -1,33 +1,253 @@
+use std::marker::PhantomData;
+
+use bevy::ecs::query::{ReadOnlyWorldQuery, WorldQuery};
 use bevy::prelude::*;
 use de_core::{baseset::GameSet, objects::ObjectType, player::Player, state::AppState};
 use de_objects::Health;
 
-use crate::{ObjectCounter, SpawnerSet};
+use crate::ObjectCounter;
 
-pub(crate) struct DestroyerPlugin;
+pub(crate) struct DespawnerPlugin;
 
-impl Plugin for DestroyerPlugin {
+impl Plugin for DespawnerPlugin {
     fn build(&self, app: &mut App) {
-        app.add_system(
-            destroy
-                .in_base_set(GameSet::Update)
-                .run_if(in_state(AppState::InGame))
-                .in_set(SpawnerSet::Destroyer),
-        );
+        app.add_event::<DespawnEvent>()
+            .add_system(
+                find_dead
+                    .in_base_set(GameSet::Update)
+                    .run_if(in_state(AppState::InGame))
+                    .before(DespawnerSet::Despawn),
```

Comment:
Shouldn't this be in `.in_set(DespawnerSet::Destruction)`?

Reply:
I think it would be better to do `.before(DespawnerSet::Despawn)` because if the health check system runs before systems mutating health (which should be running in this set) that could cause a frame of delay. 

Mabye we could do `.after(DespawnerSet::Destruction)` if that would be better?